neutralinojs/neutralinojs
# Added API method in window - captureScreen

## Description

This pull request introduces functionality for the `neutralino.window.captureScreen(filepath)` method, which captures the current application window and saves it as a screenshot to the specified file path. This functionality is fully implemented for Windows and Linux, with error handling and fallback mechanisms.

## Changes proposed

 - Implemented screen capture functionality for Windows using GDI+ with fallback to `BitBlt`.
 - Enhanced Linux support with GTK for screen capturing.
 - Added error handling and informative log messages for troubleshooting.
 - Added test suite for captureScreen method.

## How to test it

 - Run the application and call the `neutralino.window.captureScreen("screenshot.png")` function.
 - Verify the output:
   - Ensure a `screenshot.png` file is saved in the specified directory.
   - Check the content of the screenshot to ensure it reflects the application window.
 - Test on both Windows and Linux platforms:
   - **Windows**: Ensure that `PrintWindow` or `BitBlt` is used for capturing the screen.
   - **Linux**: Verify the GTK functionality for capturing the screen.
   
![Screenshot 2024-11-30 235104](https://github.com/user-attachments/assets/fe8f533c-108b-4bca-81b6-954be39b9446)

## Next steps

Extend the captureScreen functionality to support macOS.

## Deploy notes

The changes are backward-compatible and do not impact other features.

This is my first PR please provide feedback about the code quality and any other feedback.
Thanks
